src/qml/elements/ChatMessage.qml

```diff
@@ -27,43 +27,50 @@
  *  You should have received a copy of the GNU General Public License
  *  along with Kaidan.  If not, see <http://www.gnu.org/licenses/>.
  */
-
 import QtQuick 2.6
 import QtQuick.Controls 2.0 as Controls
 import org.kde.kirigami 2.0 as Kirigami
+import QtQuick.Layouts 1.3
 
-Row {
+RowLayout {
 	id: root
 	property bool sentByMe: true
 	property string messageBody: ""
 	property date dateTime: new Date()
 	property bool isRead: false
 	property string recipientAvatarUrl: ""
 
-	anchors.right: sentByMe ? parent.right : undefined
 	spacing: Kirigami.Units.gridUnit * 0.5
 	layoutDirection: sentByMe ? Qt.RightToLeft : Qt.LeftToRight
 
+	Item {
+		Layout.preferredWidth: 5
+	}
+
 	RoundImage {
 		id: avatar
 		visible: !sentByMe
-		width: Kirigami.Units.gridUnit * 2.2
-		height: Kirigami.Units.gridUnit * 2.2
 		source: recipientAvatarUrl
+		height: width
+		fillMode: Image.PreserveAspectFit
+		Layout.preferredWidth: Kirigami.Units.gridUnit * 2.2
+		Layout.preferredHeight: Kirigami.Units.gridUnit * 2.2
+		Layout.alignment: Qt.AlignHCenter | Qt.AlignTop
+		mipmap: true
 	}
 
 	Item {
 		id: bubble
 		height: label.implicitHeight + messageInfo.height
-		width: Math.max(label.width, checkmark.visible ? dateLabel.width + checkmark.width + Kirigami.Units.gridUnit * 1.7
-							       : dateLabel.width + checkmark.width + Kirigami.Units.gridUnit * 0.7)
+		width: Math.max(label.width, dateLabel.width + checkmark.width
+						+ Kirigami.Units.gridUnit * (checkmark.visible ? 1.7 : 0.7))
```

Comment:
Please use two tabs and then only spaces, to achieve the correct indentation on all tab lengths.

Reply:
Again, so far tabs are a major pain point with no reasonable profit...

Reply:
The profit is that you can use 4 char length indentation and I can use 8 char length